Feat: post api boxes

mashazyu opened the pull request:
This PR adds POST functionality for creating new boxes (devices) via the API and updates the data format returned by the /users/me/boxes endpoint to match the openSenseMap API specification.

## Changes
- Implement POST /api/boxes endpoint for creating new devices with sensors.
- Ensure that returned device data is backwards compatible for both POST /boxes and GET /users/me/boxes
- Convert last measurement data to string format (this format is expected by the app)
- I hope I fixed ts and lint issues for current pr, but I didn't address anything prior

## Testing
I have been testing current PR with senseBox:bike app.
1. After login, I get boxes data in the format supported by current version of senseBox:bike app.
2. I can create new box via app.

## Open questions
- What attributes should be returned after box is created and at /users/me/boxes? Is it ok not to return attributes with null values (like it was done before)?
- Is it ok to return last measurement as string, not as double?
- What should be returned as device.acces_token?

mashazyu commented on app/routes/api.boxes.ts:
```diff
@@ -0,0 +1,371 @@
+ *                   example: false
+ *         access_token:
+ *           type: string
+ *           description: Box access token
```
**question:** What should be returned as acces token? In other parts of code i saw that jwt token is returned in fields with similar names and did the same, but I am not sure it is correct.

scheidtdav reviewed (commented):
Regarding your open questions:
> - What attributes should be returned after box is created and at /users/me/boxes? Is it ok not to return attributes with null values (like it was done before)?

Yes thats ok. Whatever was done before is currently our goal.

> - Is it ok to return last measurement as string, not as double?

It was probably string before, so again: Yes thats ok.

> - What should be returned as device.acces_token?

I am frankly somewhat confused by this. Where does this field even come from?
Why would a device respond with an access token if the client just did a request for which a token is required. That implies the client already having a token. Do we even need this?

mashazyu commented on app/models/device.server.ts:
```diff
@@ -388,7 +391,11 @@ export async function createDevice(deviceData: any, userId: string) {
 					tags: deviceData.tags,
```
I didn't make changes to this code in current PR, therefore I will not make this change. But, if it's a useful one, please let me know, @scheidtdav

scheidtdav replied:
Its sort of useful to see that its handled, but it shouldn't make a difference.
Inserting a value of undefined will result in the default of `[]` on a sql level. Would be much better to have proper types for `createDevice` but I'd argue thats subject to a different PR. Leave this change out for now, if you ask me.

mashazyu commented on app/routes/api.users.me.boxes.ts:
```diff
@@ -20,14 +21,27 @@ export const loader: LoaderFunction = async ({
         },
       );
 
+    const rawAuthorizationHeader = request.headers.get("authorization");
+    if (!rawAuthorizationHeader) {
+      return Response.json(
+        {
+          code: "Forbidden",
+          message: "Missing authorization header",
+        },
+        { status: 403 }
+      );
+    }
```
Thank you very much for the clarification!

mashazyu commented on app/routes/api.boxes.ts:
```diff
@@ -0,0 +1,376 @@
+ *                     example: "25.13"
+ */
+
+export const action: ActionFunction = async ({
```
Thank you for pointing this out. Done!

mashazyu commented:
@scheidtdav, thank you very much for the review and feedback!

Per our conversation I removed access_token mentions and addressed your comments.

Last question - because current PR was based on some other PR, which was based on another PR, now i contains commits from one/all of those prs. How would you like me to proceed? I could create a new pr and cherry-pick only my changes? Or there is a better way?

scheidtdav commented:
@mashazyu Don't worry about those. Its a bit of a leftover from how we approached the migration previously.
Those commits are all already on dev, but because we squash and merge GitHub cannot figure out which ones to ignore for this PR. There are only your changes though :-)
